### ISSUE TITLE
platform.py conflicts with existing platform.py on Kodi 18

### ISSUE THREAD

**Issue by peno64** [Zomboided/service.vpn.manager#205]
Kodi 18 in almost there. At this moment a release candidate is available.

I upgraded my Kodi 17 to v 18 and everything is working fine, even this add-on, however there is one issue...

At a given point I wanted to install widevine from inputstreamhelper, a standard kodi add-on (/storage/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py) and this didn't work because this python script imports platform.
However because platform.py exists in your add-on, the wrong import is done. 
I think it should use /usr/lib/python2.7/platform.pyo but instead it used /storage/.kodi/addons/service.vpn.manager/libs/platform.pyo which is of course not good.

I would solve the problem by temporary renaming platform.py and platform.pyo in /storage/.kodi/addons/service.vpn.manager/libs/ into something else and then I could do the install.
However I am afraid that later on other problems will also occur because of this name clash.

So can you please change the name of your platform.py file to something else such that there is no conflict anymore?

Note that I use Kodi on Raspbery PI

Thank you

**Comment by peno64** [comment]
The script that fails is inputstreamhelper.py which is standard included in kodi.
That script is just doing 
import platform
at the beginning.
So I don't think there is something they can fix.

Why it then uses your platform.py file I don't really know. It must be because the system finds it before the standard script.

**Comment by Zomboided** [comment]
The I suspect it's a problem with the beta release not doing the right
thing with Python paths.  I'm not gonna change anything until Kodi leaves
beta tho, it's too much effort to work around what according to the Python
docs is perfectly ok (and has been through the last 2 releases of Kodi)

**Comment by peno64** [comment]
Since kodi 18 is now officially released and I see you are doing changes for v 18, I would like to reopen this ticket since the problem still occurs.

I will demonstrate the problem with a complete fresh install of kodi 18.
Note that I am doing this with LibreElec on raspberry pi.

- So first I do a fresh install of libreelec with kodi 18.
- Then install VPN Manager for OpenVPN (I just install from the latest zip from https://github.com/Zomboided/service.vpn.manager/releases/tag/5.0.7)
- When it asks to setup, I choose Settings and then Cancel. No  need to setup a vpn to show the problem
- Reboot such that the service is installed
- VPN Manager will ask again to setup a vpn. Again cancel this
- Now Widevine must be installed to show the problem. See https://en.wikipedia.org/wiki/Widevine what it is needed for. Addons like netflix need this for the newest encoding. The easiest way to install it is via the Retrospect addin. That is an addin to show the television programs of Dutch and Belgian countries. To install this addon, go to https://bitbucket.org/basrieter/xbmc-online-tv/downloads/ and download the latest zip and install this addon. 
- Go then into my addons, Video addons and there you will see Retrospect. Go into itand choose configure. In the General screen go down to  "Install Widevine using 'InputStream Helper' addon. 
- A message is given that this is at your own risk (he tells this because widevine is not developed with this repo owner but by others). Choose Yes.
- Now you get an error saying "Installation of widevine failed'.

- Now uninstall vpn manager and reboot and try above installation of widevine again and you will see that it goes further without errors. You don't have to do the whole installation, believe me it then works.

And as firstly stated, this problem is because platform.py from vpn manager is wrongly taken.

So if you could rename platform.py in your package to something else the problem is solved.
And there is a good chance that other packages would also have this problem.

I hope you are now convinced that this is really needed.

**Comment by Zomboided** [comment]
I'm not gonna install bits of random stuff that I don't use on my Kodi build, but I will test this for you.

I have VPN Mgr installed and working.  And I have another service installed called test which imports the Python platform.py and prints out the machine type.  Here's the contents of service.py which runs as service.py in test

```
import xbmc
import xbmcgui
import xbmcaddon
import xbmcvfs
import time
import platform

if __name__ == '__main__':
    monitor = xbmc.Monitor()
    
    machine = platform.machine()
    xbmc.log("Machine is " + machine, level=xbmc.LOGWARNING) 
    
    while not monitor.abortRequested():
        # Sleep/wait for abort for 10 seconds
        if monitor.waitForAbort(10):
            # Abort was requested while waiting. We should exit
            break
```

This works just fine.  It picks up the Python version of platform and prints the machine type in the log.

If I add an 'import libs.platform' to service.py, it complains that there is no localised libs.platform.  

If I create a libs/platform and fill it with a routine to return a string, I can include that from service.py and call it without issue (ie it calls the routine expected, and doesn't look in service.vpn.manager).

tl;dr  I don't think VPN Mgr is doing anything wrong.  The Python docs appears to agree.  My own testing doesn't do what you describe.  So my conclusion is that either Widevine or Kodi need to fix something.  I don't want to spend any more time on this

**Comment by peno64** [comment]
I don't report this for myself because I can and have solved this issue by renaming the file and adapting the used code. So I have a solution for myself.

But I am 100% sure that others will have this problem also. If I can simulate this with a clean install and only your addon and a library that will be needed more and more by people then you can be sure that others will have this problem also.

If you don't want to investigate this then *you* have a problem, not I

**Comment by Zomboided** [comment]
I think you've got the wrong perspective on where the problem lies.  Either way, I'm not debating this with you as you're clearly right.
